frontend/src/layout/navigation-3000/sidepanel/panels/sidePanelDocsLogic.ts

```diff
@@ -14,32 +29,66 @@ export const sidePanelDocsLogic = kea<sidePanelDocsLogicType>([
 
     actions({
         openDocsPage: (urlOrPath: string) => ({ urlOrPath }),
+        updatePath: (path: string) => ({ path }),
+        setInitialPath: (path: string) => ({ path }),
+        unmountIframe: true,
+        handleExternalUrl: (urlOrPath: string) => ({ urlOrPath }),
     }),
 
     reducers(() => ({
-        path: [
+        currentPath: [
+            null as string | null,
+            {
+                updatePath: (_, { path }) => sanitizePath(path),
+            },
+        ],
+        initialPath: [
             '/docs' as string,
+            { persist: true },
             {
-                openDocsPage: (_, { urlOrPath }) => {
-                    let path = urlOrPath
-                    try {
-                        const url = new URL(urlOrPath)
-                        if (url.origin === POSTHOG_COM_DOMAIN) {
-                            path = url.pathname + url.search
-                        }
-                    } catch (e) {
-                        // not a valid URL, continue
-                    }
-
-                    return path[0] === '/' ? path : `/${path}`
-                },
+                setInitialPath: (_, { path }) => sanitizePath(path),
             },
         ],
     })),
 
-    listeners(({ actions }) => ({
-        openDocsPage: () => {
+    selectors({
+        iframeSrc: [
+            (s) => [s.initialPath],
+            (initialPath) => {
+                return `${POSTHOG_WEBSITE_ORIGIN}${initialPath ?? ''}`
+            },
+        ],
+        currentUrl: [
+            (s) => [s.currentPath],
+            (currentPath) => {
+                return `${POSTHOG_WEBSITE_ORIGIN}${currentPath ?? ''}`
+            },
+        ],
+    }),
+
+    listeners(({ actions, values }) => ({
+        openDocsPage: ({ urlOrPath }) => {
+            actions.setInitialPath(getPathFromUrl(urlOrPath))
             actions.openSidePanel(SidePanelTab.Docs)
         },
+
+        unmountIframe: () => {
+            // Update the initialPath so that next time we load it is the same as last time
+            actions.setInitialPath(values.currentPath ?? '/docs')
+        },
+
+        handleExternalUrl: ({ urlOrPath }) => {
+            router.actions.push(getPathFromUrl(urlOrPath))
+        },
     })),
+
+    afterMount(({ actions, cache }) => {
+        // Call actions.unmountIframe when the window unloads
+        cache.beforeUnload = () => actions.unmountIframe()
+        window.addEventListener('beforeunload', cache.beforeUnload)
+    }),
+
+    beforeUnmount(({ cache }) => {
+        window.removeEventListener('beforeunload', cache.beforeUnload)
+    }),
```

Comment:
These changes may not be what you intended here. They don't fire when the tab is closed, only when the logic is mounted/unmounted. But in that case what we're doing here is already handled by the `useEffect`s in `SidePanelDocs`.

To add support for the tab being closed, we instead need to `window.addEventListener('beforeunload', beforeWindowUnloadHandler)` in the `[]`-dependent ``SidePanelDocs` `useEffect`.

Unless this is not what you meant, in which case disregard.

Reply:
Ah good point

Reply:
Fixed!